app/features-json/setup_json_controller.rb

```diff
@@ -40,5 +40,33 @@ class SetupJSONController < APIController
         }
       )
     end
+
+    post "#{HOME}/oauth", authenticate: false do
+      client_id = params["client_id"]
+      client_secret = params["client_secret"]
+
+      if client_id.nil? || client_secret.nil?
+        json_error!(
+          error_message: "Must provide both the OAuth client ID and client secret",
+          error_key: "OAuth.Missing",
+          error_code: 400
+        )
+      end
+
+      unless FastlaneCI.dot_keys.oauth_client_id.nil? || FastlaneCI.dot_keys.oauth_client_secret.nil?
```

Comment:
I'd try avoiding `unless` together with `||` and similar, as it's confusing when it comes to priorities, can we re-org this to make it easy to read, including non Ruby devs